documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_moderation_metrics.md

```diff
@@ -0,0 +1,501 @@
+    sum(case when action = 'deindexed_sensitive' then 1 else 0 end) as deindexedSensitiveCount,
+    sum(case when action = 'deindexed_copyright' then 1 else 0 end) as deindexedCopyrightCount,
+    sum(case when action = 'reversed_mark_sensitive' then 1 else 0 end) as reversedMarkSensitiveCount,
+    sum(case when action = 'reversed_deindex' then 1 else 0 end) as reversedDeindexCount,
+    sum(case when action = 'rejected_reports' then 1 else 0 end) as rejectedCount,
+    sum(case when action = 'deduplicated_reports' then 1 else 0 end) as deduplicatedCount,
+    sum(case when affected_records = 1 then 1 else 0 end) as singleCount,
+    sum(case when affected_records != 1 then 1 else 0 end) as bulkCount,
+  by bin(5m)
+| sort @timestamp desc
+```
+
+This query can be used to create a line chart that shows the volume of reports
+and our various decisions over time. Similar queries can be used to create
+charts for other metrics. Ultimately a dashboard can be put together in
+CloudWatch, that tracks all relevant metrics.
+
+This dashboard will also be tracked in the infrastructure repo for backup and
+restoration purposes.
+
+### Implement deferred metrics
+
+Deferred metrics are those that are computed periodically or on-demand from the
+data in our database. These metrics can be computed using the Django ORM and
+their results can be cached.
+
+In a future iteration, we can add functionality to dig into these values based
+on a rolling window for observing trends. For example, we can narrow down a
+metric for the last week or month.
+
+Reversal of sensitive marking or deindexing does not invalidate the accuracy of
+previous reports as they were accurate at the time of the decision and they
+still hold the `decision_id` that agrees with them.
+
+Deferred metrics can be broken down into two types, values and lists.
+
+#### Values
+
+Values are calculations that result in single numerical values. For example,
+accuracy of reports and time to decision are of this type.
+
+The following code can be used to calculate these metrics. The outcomes of these
+calculations can be cached in Redis with the desired TTL.
+
+Assume `Report`, `Decision` and `Media` here refer to the models for one media
+type. This example uses
+[package `tailslide`](https://github.com/ankane/tailslide) but a fairly basic
+aggregator could be implemented by us if needed.
+
+```python
+from datetime import datetime, timedelta
+from django.utils import timezone
+from api.models import Report, Decision, Media
+
+# setup
+# =====
+# Assume n is an integer number of days set by the user.
+n_days_ago = timezone.now() - timedelta(days=n)
+reports_in_range = Report.objects.filter(created_at__gte=n_days_ago)
+
+# accuracy of reports
+# ===================
+total_reports = reports_in_range.count()
+confirmed_reports = reports_in_range.filter(decision__action__in=['marked_sensitive', 'deindexed_sensitive', 'deindexed_copyright']).count()
```

Comment:
This is a minor thing, but the more I think about it the more the term 'accuracy' doesn't feel quite right -- specifically because it excludes deduplicated reports. We define dedeuplicating a report as an acknowledgement that it was accurate, but requires no action. (E.g. an already sensitive-marked work gets reported for sensitive content).

I totally understand why the duplication counts would be broken out into a separate metric, though. Maybe we could just change "accuracy" to something like "actionable"/"actioned"?

Reply:
Just adding that I don't think the language necessarily needs to be finalized at this stage. I would be totally fine with just mentioning that in the issues that are created.

Happy to approve whenever this is moved out of the Clarification Round!

Reply:
I tried very hard to find a better name for it but couldn't, because of these reaosons

- duplicates can exist for both accurate and inaccurate reports so "accuracy" isn't the best term for this
- marking a report as a duplicate is an action so not counting them under "actionable"/"actioned" isn't fair either

I'll move this proposal ahead into the decision round if we can keep this as an open question to find a better, more-suitable name for this metric.